src/codegen/mod.rs

```diff
@@ -2192,7 +2192,9 @@ impl CodeGenerator for CompInfo {
                                         quote! {
                                             assert_eq!(
                                                 unsafe {
-                                                    &(*(::#prefix::ptr::null::<#canonical_ident>())).#field_name as *const _ as usize
+                                                    let uninit = ::#prefix::mem::MaybeUninit::<#canonical_ident>::uninit();
+                                                    let ptr = uninit.as_ptr();
+                                                    ::#prefix::ptr::addr_of!((*ptr).#field_name) as usize - ptr as usize
```

Comment:
using offset_from() might work here (as `#field_offset` will be a literal you don't need to worry about usize/isize conversions)

Reply:
`offset_from()` assumes that the two pointers have the same type, which is not the case in this case, and if we were to cast to `*const u8`, I would rather just directly use `usize`.